imblearn/pipeline.py

```diff
@@ -215,22 +215,15 @@ def _fit(self, X, y=None, **fit_params):
                 with _print_elapsed_time('Pipeline',
                                          self._log_message(step_idx)):
                     continue
-            if hasattr(memory, "location"):
+
+            try:
                 # joblib >= 0.12
-                if memory.location is None:
-                    # we do not clone when caching is disabled to
-                    # preserve backward compatibility
-                    cloned_transformer = transformer
-                else:
-                    cloned_transformer = clone(transformer)
-            elif hasattr(memory, "cachedir"):
-                # joblib <= 0.11
-                if memory.cachedir is None:
-                    # we do not clone when caching is disabled to
-                    # preserve backward compatibility
-                    cloned_transformer = transformer
-            else:
-                cloned_transformer = clone(transformer)
+                mem = memory.location
+            except Exception:
```

Comment:
Could you check for the real Exception type because we are catching any type.

Reply:
Yes, of course. But I thought that in this case the only exception that can occur is the `AttributeError`. Usually you define different exception handling if different kind of exceptions may arise. But I couldn't thought an exception apart of the one I mentioned. I may be wrong, though.

Reply:
I think this is fine to have `AttributeErrror`.

Reply:
Lets to this in https://github.com/scikit-learn-contrib/imbalanced-learn/pull/687/commits/56d0da3deade34339029019c8f0a0dccf2768e21